Assets/MRTK/SDK/Features/Utilities/Solvers/HandConstraintPalmUp.cs

```diff
@@ -163,5 +203,209 @@ protected override bool IsValidController(IMixedRealityController controller)
+        {
+            Ray gazeRay;
+
+            bool usedEyeGaze = InputRayUtils.TryGetRay(InputSourceType.Eyes, Handedness.Any, out gazeRay);
+
+            if (usedEyeGaze || InputRayUtils.TryGetRay(InputSourceType.Head, Handedness.Any, out gazeRay))
+            {
+                // Define the activation point as a vector between the wrist and pinky knuckle; then cast it against the plane to get a smooth location
+                Vector3 activationPoint;
+                Plane handPlane;
+                float distanceToHandPlane;
+
+                // If we can generate the handplane/are able to set an activation point on it, and then are able to raycast against it
+                if (GenerateHandPlaneAndActivationPoint(jointedHand, out handPlane, out activationPoint) && 
+                    handPlane.Raycast(gazeRay, out distanceToHandPlane))
+                {
+                        // Now that we know the dist to the plane, create a vector at that point
+                        Vector3 gazePosOnPlane = gazeRay.origin + gazeRay.direction.normalized * distanceToHandPlane;
+                        Vector3 planePos = handPlane.ClosestPointOnPlane(gazePosOnPlane);
+                        float gazePosDistToActivationPosition = (activationPoint - planePos).sqrMagnitude;
+                        float gazeActivationThreshold = usedEyeGaze ? eyeGazeProximityThreshold : headGazeProximityThreshold;
+                        bool gazeActivated = gazeActivationAlreadyTriggered = (gazePosDistToActivationPosition < gazeActivationThreshold);
+
+                        return gazeActivated;
+                    }
+                }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Coroutine function called by the ManipulationHandler of the attached object whenever the object is done 
+        /// being manipulated by the user. This triggers a coroutine that checks to see whether the object should 
+        /// reattach to the hand.
+        /// </summary>
+        public void StartWorldLockReattachCheckCoroutine()
+        {
+            StartCoroutine(WorldLockedReattachCheck());
+        }
+
+        /// <summary>
+        /// This function attenpts to generate a hand plane based on the wrist, index knuckle and pinky knuckle joints present in the hand.
+        /// On a success, it then calls GenerateActivationPoint to try to generate a hand-based activation point that the user
+        /// needs to gaze at to activate the constrained object.
+        /// On a failure, it assigns them to be default values and then returns false
+        /// </summary>
+        /// <param name="jointedHand"></param>
+        /// <param name="handPlane"></param>
+        /// <param name="activationPoint"></param>
+        /// <returns></returns>
+        private bool GenerateHandPlaneAndActivationPoint(IMixedRealityHand jointedHand, out Plane handPlane, out Vector3 activationPoint)
```

Comment:
Would be nice to rename to `Try GenerateHandPlaneAndActivationPoint` to follow the pattern used elsewhere in MRTK to use Try* when calling methods that could fail.

Reply:
Sure; can do.